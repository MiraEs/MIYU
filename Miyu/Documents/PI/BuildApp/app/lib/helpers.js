import moment from 'moment';
import numeral from 'numeral';
import {
	findNodeHandle,
	PixelRatio,
	StatusBar,
	Linking,
	Platform,
	CameraRoll,
	ImageEditor,
	NativeModules,
} from 'react-native';
import {
	SITE_ID,
	CLOUDINARY_URL,
} from '../constants/constants';
import { AllHtmlEntities } from 'html-entities';
import NetworkError from '../errors/NetworkError';
const { UIManager } = NativeModules;

const entities = new AllHtmlEntities();
let endCursor = null;

const helpers = {

	isIOS() {
		return Platform.OS === 'ios';
	},

	isAndroid() {
		return Platform.OS === 'android';
	},

	noop() {},

	/**
	 * Get image data for a given URI
	 * This is a recursive function so be careful modifying it
	 */
	getPhotoByUri(photoUri) {
		const options = {
			first: 1000,
		};
		if (endCursor) {
			options.after = endCursor;
		}
		return CameraRoll.getPhotos(options).then((data) => {
			if (data && data.edges && data.edges.length) {
				const match = data.edges.find((edge) => edge.node.image.uri === photoUri);
				if (match) {
					endCursor = null;
					return match.node.image;
				} else if (data.page_info && data.page_info.has_next_page) {
					endCursor = data.page_info.end_cursor;
					return helpers.getPhotoByUri(photoUri);
				}
			}
		});
	},

	getResizedDimensions(width, height, maxWidth, maxHeight) {
		if (height > width && height > maxHeight) {
			width = (maxHeight / height) * width;
			height = maxHeight;
		} else if (width > height && width > maxWidth) {
			height = (maxWidth / width) * height;
			width = maxWidth;
		}
		return {
			width: helpers.toInteger(width),
			height: helpers.toInteger(height),
		};
	},

	/**
	 * Resize the given images
	 */
	resizeImages(images = []) {
		const resizeList = [];
		if (Array.isArray(images)) {
			images.forEach((image) => {
				resizeList.push(new Promise((resolve, reject) => {
					return helpers.getPhotoByUri(image.uri).then((imageData) => {
						if (imageData) {
							const { width, height } = imageData;
							const dimensions = helpers.getResizedDimensions(width, height, image.maxWidth, image.maxHeight);
							ImageEditor.cropImage(image.uri, {
								offset: {
									x: 0,
									y: 0,
								},
								size: {
									width,
									height,
								},
								displaySize: {
									width: dimensions.width,
									height: dimensions.height,
								},
							}, resolve, reject);
						} else {
							reject();
						}
					});
				}));
			});
		}
		return Promise.all(resizeList);
	},

	findUs() {
		const url = 'geo:39.7077069,-121.8205423,19.78';
		Linking
			.canOpenURL(url)
			.then((supported) => {
				if (supported) {
					Linking.openURL(url);
				}
			});
	},

	emailUs(email) {
		const url = `mailto:${email}`;
		Linking
			.canOpenURL(url)
			.then((supported) => {
				if (supported) {
					Linking.openURL(url);
				}
			});
	},

	getCloudinaryImageUrl(options) {
		if (options.name) {
			return `http://s3.img-b.com/image/private/t_base,f_auto,dpr_2.0,c_${options.crop || 'lpad'}${options.gravity ? `,g_${options.gravity}` : ''}${options.width ? `,w_${Math.round(options.width)}` : ''}${options.height ? `,h_${options.height}` : ''}${options.effect ? `,e_${options.effect}` :''}/${options.section ? `${options.section}/` : ''}${options.manufacturer ? `${options.manufacturer.toLowerCase().replace(/\s/g, '')}/` : ''}${options.name.toLowerCase().replace(/\s/g, '')}`;
		}
		return helpers.getCloudinaryNoImageAvailableUrl(options);
	},

	/**
	 * Get the no image available URL from cloudinary at the specified size
	 * @param  {object} options height and width
	 * @return {string}         the no image available URL
	 */
	getCloudinaryNoImageAvailableUrl(options) {
		return `https://s3.img-b.com/image/upload/t_base,f_auto,dpr_2.0,c_lpad,w_${options.width},h_${options.height}/404images/noimage.gif`;
	},

	getEventTitle(eventType) {
		switch (eventType) {
			case 'POST':
				return 'Post';
			case 'TEAM_MEMBER':
				return 'Team Member';
			case 'TEAM_EXPERT':
				return 'Team Expert';
			case 'FAVORITE_LIST':
				return 'Favorites List';
			case 'ORDER':
				return 'Order';
			case 'TRACKING':
				return 'Tracking';
			default:
				return '';
		}
	},

	getCloudinarySalesRepImage(options) {
		return `https://s3.img-b.com/image/upload/t_base,w_${options.width},h_${options.height}/v1/mediabase/build_profiles/${options.repUserId}/profile/${options.repUserId}`;
	},

	getProfileImage(options) {
		return `https://s3.img-b.com/image/upload/t_profile,w_${options.width || 200},h_${options.height || 200},c_thumb,dpr_2.0/v1/mediabase/build_profiles/${options.repUserId}/profile/${options.repUserId}`;
	},

	slugify(text) {
		if (text) {
			// Replace spaces with -
			// Remove all non-word chars including underscores
			// Replace multiple - with single -
			// Trim - from start of text
			// Trim - from end of text
			return text
				.toLowerCase()
				.replace(/\s+/g, '-')
				.replace(/([^\w\-]|_)+/g, '-')
				.replace(/\-\-+/g, '-')
				.replace(/^-+/, '')
				.replace(/-+$/, '');
		} else {
			return text;
		}
	},

	removeHTML(html) {
		return entities.decode((html || '')
			.replace(/<(?:.|\n)*?>/gm, '')
			.replace(/[\n\r]+/g, '')
			.replace(/[\t]+/g, ' '))
			.replace(/&#x27;+/g, '\'')
			.trim();
	},

	escapeRegex(text) {
		return text.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&');
	},

	getImageUrl(manufacturer, image) {
		if (image && manufacturer) {
			manufacturer = manufacturer.toLowerCase().replace(/\s/g, '');

			return `http://s2.img-b.com/imagebase/resized/330x320/${manufacturer}images/${image.toLowerCase()}`;
		}
		return 'http://s2.img-b.com/imagebase/resized/100x100/404images/noimage.gif';
	},

	//@Offload
	getCategoryImageUrl(image) {
		if (image) {
			return `http://s2.img-b.com/build.com/imagebase/resized/140x140/category_${SITE_ID}/${image.toLowerCase()}`;
		}
		return 'http://s2.img-b.com/imagebase/resized/100x100/404images/noimage.gif';
	},

	getOverlayImageUrl(id, height) {
		return 	`https://s3.img-b.com/image/private/t_base,f_png,h_${height},dpr_2.0/${id}`;
	},

	getPricedOptionUrl(image) {
		if (image) {
			return `https://s2.img-b.com/build.com/mediabase/pricedoptionimages/${image.toLowerCase()}`;
		}
		return 'https://s2.img-b.com/imagebase/resized/100x100/404images/noimage.gif';
	},

	getIcon(name) {
		return `${helpers.isIOS() ? 'ios-' : 'md-'}${name}`;
	},

	setStatusBarStyle(style, animate) {
		if (helpers.isIOS()) {
			StatusBar.setBarStyle(style, animate);
		}
	},

	/**
	 * Set visibility of status bar
	 * Animation options are 'none', 'fade', and 'slide'
	 */
	setStatusBarHidden(hidden, animation) {
		if (helpers.isIOS()) {
			StatusBar.setHidden(hidden, animation);
		}
	},

	getResizedImageForUrl(url, width) {
		if (url && url.indexOf(CLOUDINARY_URL) !== -1) {
			const index = url.lastIndexOf('/');
			// we call PixelRatio from React here because I can't seem to figure out
			// how to get Babel to destructure PixelRatio correctly in the tests
			width = width * PixelRatio.get();
			return `${url.slice(0, index)}/w_${Math.round(width)}/${url.slice(index + 1)}`;
		} else {
			return url;
		}
	},

	getDate(date) {
		const d = new Date();
		const threshold1 = d.setDate(d.getDate() - 5);
		const threshold2 = d.setDate(d.getDate() - 30);
		if (date > threshold1) {
			return moment(date).startOf('minute').fromNow();
		} else if (date > threshold2) {
			return moment(date).format('MMM D');
		} else {
			return moment(date).format('MMM D, YYYY');
		}
	},

	getRelativeDate(date) {
		const d = new Date();
		const threshold1 = d.setDate(d.getDate() - 5);
		const threshold2 = d.setDate(d.getDate() - 30);
		if (date > threshold1) {
			return moment(date).startOf('day').fromNow();
		} else if (date > threshold2) {
			return moment(date).startOf('month').fromNow();
		} else {
			return moment(date).startOf('year').fromNow();
		}
	},

	getFormattedDate(date, format) {
		return moment(new Date(date)).format(format);
	},

	getDateStrictFormat(date) {
		if (!date) {
			return;
		}
		return moment(date).format('MM/DD/YYYY');
	},
	validFinish(finish) {
		return (finish.status && finish.status.toLowerCase() === 'stock'
				&& finish.image);
	},
	setFirstAvailableFinish(productDrop) {
		const { searchRelevantFinishIndex, finishes } = productDrop,
			getFirstViableFinish = (finishes) => {
				let index = 0;
				for (const finish of finishes) {
					if (helpers.validFinish(finish)) {
						return index;
					}
					index++;
				}
				return 0;
			};


		if (searchRelevantFinishIndex && helpers.validFinish(finishes[searchRelevantFinishIndex])) {
			productDrop.selectedFinishIndex = searchRelevantFinishIndex;
		} else {
			productDrop.selectedFinishIndex = getFirstViableFinish(finishes);
		}

		return productDrop;
	},

	setLowestPrice(productDrop) {
		productDrop.minPrice = productDrop.finishes[0].cost;
		for (const finish of productDrop.finishes) {
			if (finish.status && finish.status.toLowerCase() === 'stock' && finish.cost < productDrop.minPrice) {
				productDrop.minPrice = finish.cost;
			}
		}
		return productDrop;
	},

	toInteger(number) {
		const parsedNumber = Number.parseInt(number);
		return Number.isNaN(parsedNumber) ? 0 : parsedNumber;
	},

	toFloat(number) {
		const parsedNumber = Number.parseFloat(number);
		return Number.isNaN(parsedNumber) ? 0.0 : parsedNumber;
	},

	/**
	 * Numeral.js wrapped methods
	 */
	toUSD(number) {
		if (!Number.isNaN(number) && typeof number !== 'undefined') {
			return numeral(number).format('$0,0.00');
		}
	},

	toQuantity(number) {
		return numeral(number).format('0');
	},

	toBigNumber(number) {
		return numeral(number).format('0,0');
	},
	/**
	 * Numeral.js wrapped methods
	 */
	toSqFt(number) {
		if (typeof number !== 'undefined') {
			return numeral(number).format('0,0.00');
		}
	},
	autoCapitalize(string = '', all = false) {
		if (!string || typeof string !== 'string') {
			return '';
		}
		if (all) {
			return string.replace(/(^|[^a-zA-Z\u00C0-\u017F'])([a-zA-Z\u00C0-\u017F])/g, (m) => m.toUpperCase());
		} else {
			return string.charAt(0).toUpperCase() + string.substring(1);
		}
	},

	getCouponCodeFromCart(cart) {
		if (cart.couponCode && cart.couponCode.code) {
			return cart.couponCode.code;
		}
	},

	calcGrandTotal(cart, credit) {
		const { cart: { subTotal, couponTotal, shippingOptions, taxAmount }, selectedShippingIndex } = cart;
		let grandTotal = couponTotal ? subTotal - couponTotal : subTotal;

		if (shippingOptions && shippingOptions[selectedShippingIndex] && shippingOptions[selectedShippingIndex].shippingCost) {
			grandTotal += shippingOptions[selectedShippingIndex].shippingCost;
		}

		if (taxAmount) {
			grandTotal += taxAmount;
		}

		if (credit) {
			grandTotal = credit > grandTotal ? 0 : grandTotal - credit;
		}

		return numeral(grandTotal).format('0.00');
	},

	calcStoreCredit(cart, credit) {
		const grandTotal = parseFloat(this.calcGrandTotal(cart));
		const storeCredit = credit > grandTotal ? grandTotal : credit;

		return parseFloat(storeCredit);
	},

	serviceErrorCheck(results) {
		const error = Array.isArray(results) ? results[0] : results;

		if (error && error.code && error.message) {
			throw new Error(error.message);
		}
	},

	isNetworkError(error) {
		return error instanceof NetworkError;
	},

	isGeneralError(error) {
		return !this.isNetworkError(error);
	},

	isPushNotificationsAllowed({ badge, sound, alert }) {
		return !!badge || !!sound || !!alert;
	},

	measureComponentByRef(ref, onMeasure) {
		const handle = findNodeHandle(ref);
		UIManager.measure(handle, (x, y, width, height, pageX, pageY) => {
			onMeasure({ width, height, pageX, pageY });
		});
	},

};

module.exports = helpers;
