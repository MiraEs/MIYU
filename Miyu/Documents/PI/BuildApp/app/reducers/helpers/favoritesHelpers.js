'use strict';
import EventEmitter from '../../lib/eventEmitter';

const favoritesHelpers = {
	productInFavorites(compositeId, favorites) {
		const favoriteIds = [];
		Object.keys(favorites).forEach((favoriteId) => {
			const productMap = favorites[favoriteId].productsMap;
			Object.keys(productMap).forEach((currentCompositeId) => {
				if (parseInt(currentCompositeId, 10) === compositeId) {
					favoriteIds.push(favoriteId);
				}
			});
		});
		return favoriteIds;
	},

	isProductFavorite(compositeId, favorites) {
		return favoritesHelpers.productInFavorites(compositeId, favorites).length > 0;
	},
	showFavoritesListSelector(favorites, compositeId, onDonePress, onCreateListPress) {
		const favoriteIds = Object.keys(favorites);
		const options = favoriteIds.map((key) => {
			return {
				text: favorites[key].name,
			};
		});
		const selections = favoritesHelpers.productInFavorites(compositeId, favorites).map((favoriteId) => favoriteIds.indexOf(favoriteId) + 1);
		EventEmitter.emit('showActionSheet', {
			optionsTextStyle: { textAlign: 'left' },
			title: 'Select Favorites',
			multiSelect: true,
			createNewOption: {
				text: 'Create new favorites...',
				onPress: onCreateListPress,
			},
			onDonePress,
			options,
			selections,
		});
	},
};

module.exports = favoritesHelpers;
