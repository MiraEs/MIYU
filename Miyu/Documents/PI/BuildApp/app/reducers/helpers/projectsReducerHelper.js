'use strict';

export function hasProjects(projects = {}) {
	if (!Object.keys(projects).length) {
		return false;
	}
	const {
		active: {
			invitedProjects: activeInvited = [],
			myProjects: activeProjects = [],
			sharedProjects: activeShared = [],
		},
		archived: {
			invitedProjects: archivedInvited = [],
			myProjects: archivedProjects = [],
			sharedProjects: archivedShared = [],
		},
	} = projects;

	const hasActive = activeProjects.length > 0 ||
		activeShared.length > 0 ||
		activeInvited.length > 0;
	const hasArchived = archivedProjects.length > 0 ||
		archivedShared.length > 0 ||
		archivedInvited.length > 0;

	return hasActive || hasArchived;
}

export function hasMergedProjects(projects = {}, min = 0) {
	if (!Object.keys(projects).length) {
		return false;
	}
	const {
		active: {
			invitedProjects: activeInvited = [],
			myProjects: activeProjects = [],
			sharedProjects: activeShared = [],
		},
		archived: {
			invitedProjects: archivedInvited = [],
			myProjects: archivedProjects = [],
			sharedProjects: archivedShared = [],
		},
	} = projects;
	return (activeProjects.length + activeShared.length + activeInvited.length) > min ||
		(archivedProjects.length + archivedShared.length + archivedInvited.length) > min;
}

export function getProject(projects, id) {
	const allProjects = [].concat(
		projects.active.myProjects,
		projects.active.sharedProjects,
		projects.active.invitedProjects,
		projects.archived.myProjects,
		projects.archived.sharedProjects,
		projects.archived.invitedProjects
	);
	const project = allProjects.filter((project = {}) => {
		return project.id === id;
	});
	if (project.length === 0) {
		return {};
	}
	return project[0];
}

export function findIndexOfSessionCartInShoppingList(allShoppingListsInProject, projectShoppingListGroupId) {
	if (!allShoppingListsInProject) {
		return -1;
	}
	let index = 0;
	let list;
	const { length } = allShoppingListsInProject;
	for (; index < length; index++) {
		list = allShoppingListsInProject[index];
		if (list && list.projectShoppingListGroupId === projectShoppingListGroupId) {
			break;
		}
	}

	return index === length ? -1 : index;
}

export function findIndexOfShoppingListsInProject(allShoppingLists, projectId) {
	if (!allShoppingLists) {
		return -1;
	}
	let index = 0;
	let list;
	const { length } = allShoppingLists;
	for (; index < length; index++) {
		list = allShoppingLists[index];
		if (list && list.project && list.project.id === projectId) {
			break;
		}
	}

	return index === length ? -1 : index;
}

export function getShoppingListsFromProject(allShoppingLists, projectId) {
	const index = findIndexOfShoppingListsInProject(allShoppingLists, projectId);
	return index > -1 ? allShoppingLists[index].shoppingLists || [] : [];
}

export function mapProjectsDataToObject(projects) {
	const {
		active: activeProjects,
		archived: archivedProjects,
	} = projects;
	const activeData = [];
	const archivedData = [];

	if (activeProjects.myProjects && activeProjects.myProjects.length) {
		activeData.push(...activeProjects.myProjects);
	}
	if (activeProjects.sharedProjects && activeProjects.sharedProjects.length) {
		activeData.push(...activeProjects.sharedProjects);
	}
	if (activeProjects.invitedProjects && activeProjects.invitedProjects.length) {
		activeData.push(
			...(activeProjects.invitedProjects.map((project) => ({
				...project,
				isInvited: true,
			})))
		);
	}

	if (archivedProjects.myProjects && archivedProjects.myProjects.length) {
		archivedData.push(...archivedProjects.myProjects);
	}
	if (archivedProjects.sharedProjects && archivedProjects.sharedProjects.length) {
		archivedData.push(...archivedProjects.sharedProjects);
	}
	if (archivedProjects.invitedProjects && archivedProjects.invitedProjects.length) {
		archivedData.push(
			...(archivedProjects.invitedProjects.map((project) => ({
				...project,
				isInvited: true,
			})))
		);
	}

	return {
		active: activeData,
		archived: archivedData,
	};
}
