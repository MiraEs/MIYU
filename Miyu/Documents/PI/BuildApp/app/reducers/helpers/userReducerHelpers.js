'use strict';

export function getAvatarUrl(user) {
	if (user && user.apiUser && user.apiUser.avatar) {
		return user.apiUser.avatar;
	}
}

export function getFullName(user) {
	if (user.firstName && user.lastName) {
		return `${user.firstName} ${user.lastName}`;
	} else if (user.firstName) {
		return user.firstName;
	}
}
