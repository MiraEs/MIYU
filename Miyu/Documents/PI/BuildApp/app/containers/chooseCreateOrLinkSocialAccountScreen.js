'use strict';

import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	View,
	StyleSheet,
} from 'react-native';
import {
	Button,
	Text,
} from 'BuildLibrary';
import OrSeparator from '../components/OrSeparator';
import styles from '../lib/styles';
import {
	createSocialLoginCustomer,
	updatePresentNoEmailError,
} from '../actions/SocialLoginActions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import TrackingActions from '../lib/analytics/TrackingActions';
import helpers from '../lib/helpers';
import tracking from '../lib/analytics/tracking';
import {
	trackAction,
	trackState,
} from '../actions/AnalyticsActions';

const componentStyles = StyleSheet.create({
	middle: {
		marginVertical: styles.measurements.gridSpace4,
	},
	buttons: {
		flex: 0,
		marginTop: styles.measurements.gridSpace3,
	},
});

export class ChooseCreateOrLinkSocialAccountScreen extends Component {

	constructor(props) {
		super(props);
		this.props.actions.trackState('build:app:createOrLinkSocialAccount');
		this.state = {
			isCreateAccountLoading: false,
		};
	}

	createAccount = () => {
		const {
			actions: { createSocialLoginCustomer },
		} = this.props;

		this.setState({
			isCreateAccountLoading: true,
		});

		createSocialLoginCustomer().then((data) => {
			if (data && data.shouldPromptForCredentials) {
				// if we're here, we've detected that the user already has an account with that email
				// so we weren't able to create their account and they need to link instead
				this.props.navLogin(true);
			} else {
				tracking.trackAccountCreated();
				const { actions, cart, user } = this.props;
				actions.trackAction(TrackingActions.CUSTOMER_SIGNUP_COMPLETE, {
					user,
					cart,
					methodName: 'Facebook',
				});
				this.props.loginSuccess();
			}

			this.setState({
				isCreateAccountLoading: false,
			});
		})
		.catch(() => {
			this.setState({
				isCreateAccountLoading: false,
			});
		});
	};

	renderNoEmailError = () => {
		const {
			user: { noEmailErrorMessage },
		} = this.props;

		if (noEmailErrorMessage) {
			return <Text color="error">{noEmailErrorMessage}</Text>;
		}
	};

	render() {
		const buttonsDisabled = !!this.props.user.noEmailErrorMessage;

		return (
			<View>
				{this.renderNoEmailError()}
				<Text
					size="large"
					weight="bold"
					textAlign="center"
				>
					Do you want to create a Build.com account?
				</Text>
				<Button
					key="createAccount"
					disabled={buttonsDisabled}
					isLoading={this.state.isCreateAccountLoading}
					onPress={this.createAccount}
					text="Create My Account"
					style={componentStyles.buttons}
					accessibilityLabel="Create Account"
					trackAction={TrackingActions.CREATE_SOCIAL_ACCOUNT}
				/>
				<OrSeparator style={componentStyles.middle} />
				<Text
					size="large"
					weight="bold"
					textAlign="center"
				>
					Already have a Build.com account?
				</Text>
				<Text textAlign="center">
					Log in to link your Build and Facebook accounts
				</Text>
				<Button
					key="login"
					disabled={buttonsDisabled}
					onPress={this.props.navLogin}
					text="Login"
					style={componentStyles.buttons}
					color="white"
					textColor="secondary"
					accessibilityLabel="Login"
					trackAction={TrackingActions.LINK_SOCIAL_ACCOUNT}
				/>
			</View>
		);
	}

}

ChooseCreateOrLinkSocialAccountScreen.propTypes = {
	actions: PropTypes.object.isRequired,
	cart: PropTypes.object,
	loginSuccess: PropTypes.func.isRequired,
	navCreate: PropTypes.func.isRequired,
	navLogin: PropTypes.func.isRequired,
	user: PropTypes.object,
};

ChooseCreateOrLinkSocialAccountScreen.defaultProps = {
	loginSuccess: helpers.noop,
};

const mapStateToProps = (state) => {
	return {
		user: state.userReducer.user,
		cart: state.cartReducer.cart,
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		actions: bindActionCreators({
			createSocialLoginCustomer,
			updatePresentNoEmailError,
			trackAction,
			trackState,
		}, dispatch),
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(ChooseCreateOrLinkSocialAccountScreen);
