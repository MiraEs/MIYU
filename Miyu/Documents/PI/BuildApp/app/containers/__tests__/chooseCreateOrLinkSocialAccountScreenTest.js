'use strict';

jest.mock('../../../app/services/httpClient', () => ({}));
jest.mock('../../../app/store/configStore', () => ({}));
jest.mock('../../../app/lib/analytics/tracking');
jest.mock('BuildNative');
jest.mock('BuildLibrary');
jest.mock('../../../app/components/Form', () => 'Form');
jest.mock('../../../app/components/FormInput', () => 'FormInput');

jest.unmock('react-native');

import { ChooseCreateOrLinkSocialAccountScreen } from '../chooseCreateOrLinkSocialAccountScreen';
import React from 'react';
import helpers from '../../lib/helpers';

const defaultProps = {
	actions: {
		trackState: jest.fn(),
	},
	loginSuccess: helpers.noop,
	user: {
		noEmailErrorMessage: false,
	},
};

describe('ChooseCreateOrLinkSocialAccountScreen component', () => {

	it('should render correctly', () => {
		const tree = require('react-test-renderer').create(
			<ChooseCreateOrLinkSocialAccountScreen {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
