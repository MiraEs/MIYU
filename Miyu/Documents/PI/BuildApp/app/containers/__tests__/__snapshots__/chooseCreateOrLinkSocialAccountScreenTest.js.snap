// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseCreateOrLinkSocialAccountScreen component should render correctly 1`] = `
<View>
  <Text
    size="large"
    textAlign="center"
    weight="bold"
  >
    Do you want to create a Build.com account?
  </Text>
  <Button
    accessibilityLabel="Create Account"
    disabled={false}
    isLoading={false}
    onPress={[Function]}
    style={
      Object {
        "flex": 0,
        "marginTop": 24,
      }
    }
    text="Create My Account"
    trackAction="build:app:createOrLinkSocialAccount:create"
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        Object {
          "marginVertical": 32,
        },
      ]
    }
  >
    <Text
      lineHeight={false}
      style={
        Object {
          "flex": 1,
        }
      }
      textAlign="center"
    >
      OR
    </Text>
  </View>
  <Text
    size="large"
    textAlign="center"
    weight="bold"
  >
    Already have a Build.com account?
  </Text>
  <Text
    textAlign="center"
  >
    Log in to link your Build and Facebook accounts
  </Text>
  <Button
    accessibilityLabel="Login"
    color="white"
    disabled={false}
    onPress={undefined}
    style={
      Object {
        "flex": 0,
        "marginTop": 24,
      }
    }
    text="Login"
    textColor="secondary"
    trackAction="build:app:createOrLinkSocialAccount:link"
  />
</View>
`;
